tiny_dnn/config.h

```diff
@@ -31,6 +31,11 @@
 // #define CNN_USE_OMP
 
 /**
+ * define to enable Grand Central Dispatch parallelization
+ */
+//#define CNN_USE_GCD
```

Comment:
This PR is only meant to add another optional flavour of parallel_for that uses GCD, much like the other already existing flavours like CNN_USE_OMP and CNN_USE_TBB above in the same file. I just added this in the same style. 

Maybe there is a better way?

Adding it but not commenting it out would enable the GCD version for all platforms and that wouldn't work.

It could be argued that CNN_USE_GCD would be a good default option on macOS so that people who just try tiny-dnn out on a Mac don't get the slower default implementation.